### PR TITLE
Bug 2031564: Separate out test for 'RequiredInstallerResourcesMissing secrets'

### PR DESCRIPTION
**Pull request by DennisPeriquet**
Part of [TRT-159](https://issues.redhat.com/browse/TRT-159) and related to [OCPBUGSM-38392](https://issues.redhat.com/browse/OCPBUGSM-38392) and [BZ 2031564](https://bugzilla.redhat.com/show_bug.cgi?id=2031564).

### CODE REVIEW

File: pkg/synthetictests/image_pulls.go [hunk 1]
```diff
@@ -53,9 +55,6 @@ func (s *singleEventCheckRegex) test(events monitorapi.Intervals) []*junitapi.JU
 	}
 	if len(failureOutput) > 0 {
 		totalOutput := failureOutput
-		if len(flakeOutput) >= 0 {
```
Comment (DennisPeriquet) [inline pkg/synthetictests/image_pulls.go:56]: This if statement is harmless but doesn't need to be here.

File: pkg/synthetictests/image_pulls.go [hunk 2]
```diff
@@ -96,5 +97,14 @@ func newSingleEventCheckRegex(testName, regex string) *singleEventCheckRegex {
...
-	return newSingleEventCheckRegex(testName, imagePullRedhatRegEx).test(events)
+	return newSingleEventCheckRegex(testName, imagePullRedhatRegEx, math.MaxInt, imagePullRedhatFlakeThreshold).test(events)
+}
+
+// testRequiredInstallerResourcesMissing looks for this symptom:
+// and fails if it happens more than the failure threshold count of 20 and flakes more than the
+// flake threshold.  See https://bugzilla.redhat.com/show_bug.cgi?id=2031564.
+//
+func testRequiredInstallerResourcesMissing(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	testName := "[sig-etcd] should not see excessive RequiredInstallerResourcesMissing secrets"
```
Comment (deads2k) [inline pkg/synthetictests/image_pulls.go:108]: bz-etcd

File: pkg/synthetictests/image_pulls.go [hunk 3]
```diff
@@ -11,8 +11,11 @@ import (
 )
 
 const (
-	imagePullRedhatRegEx          = `reason/[a-zA-Z]+ .*Back-off pulling image .*registry.redhat.io`
-	imagePullRedhatFlakeThreshold = 5
+	imagePullRedhatRegEx                    = `reason/[a-zA-Z]+ .*Back-off pulling image .*registry.redhat.io`
+	imagePullRedhatFlakeThreshold           = 5
+	requiredresourcesMissingRegEx           = `reason/RequiredInstallerResourcesMissing secrets: etcd-all-certs-[0-9]+`
+	requiredResourceMissingFlakeThreshold   = 10
+	requiredResourceMissingFailureThreshold = 20
```
Comment (deads2k) [inline pkg/synthetictests/image_pulls.go:18]: use the existing failure threshold from the primary test.

File: pkg/synthetictests/image_pulls.go [hunk 4]
```diff
@@ -96,5 +96,14 @@ func newSingleEventCheckRegex(testName, regex string) *singleEventCheckRegex {
 //
 func testBackoffPullingRegistryRedhatImage(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
 	testName := "[sig-arch] should not see excessive pull back-off on registry.redhat.io"
-	return newSingleEventCheckRegex(testName, imagePullRedhatRegEx).test(events)
+	return newSingleEventCheckRegex(testName, imagePullRedhatRegEx, math.MaxInt, imagePullRedhatFlakeThreshold).test(events)
+}
+
+// testRequiredInstallerResourcesMissing looks for this symptom:
```
Comment (dgoodwin) [inline pkg/synthetictests/image_pulls.go:102]: Looks like you meant to paste the symptom here.

File: pkg/synthetictests/image_pulls.go [hunk 5]
```diff
@@ -11,8 +11,10 @@ import (
 )
 
 const (
-	imagePullRedhatRegEx          = `reason/[a-zA-Z]+ .*Back-off pulling image .*registry.redhat.io`
-	imagePullRedhatFlakeThreshold = 5
+	imagePullRedhatRegEx                  = `reason/[a-zA-Z]+ .*Back-off pulling image .*registry.redhat.io`
+	imagePullRedhatFlakeThreshold         = 5
+	requiredresourcesMissingRegEx         = `reason/RequiredInstallerResourcesMissing secrets: etcd-all-certs-[0-9]+`
```
Comment (dgoodwin) [inline pkg/synthetictests/image_pulls.go:16]: should be requiredResourcesMIssingRegEx

File: pkg/synthetictests/image_pulls.go (hunk 4 above)
Comment (DennisPeriquet) [reply pkg/synthetictests/image_pulls.go:102]: you're right.  Fixed

File: pkg/synthetictests/image_pulls.go (hunk 5 above)
Comment (DennisPeriquet) [reply pkg/synthetictests/image_pulls.go]: good 👁️

**Comment by dgoodwin** [comment]
/bugzilla refresh

@DennisPeriquet this needs a conflict resolve.

**Pull request merged**